src/views/SearchResults.js

```diff
@@ -0,0 +1,241 @@
+      this.searchArticle(searchQuery);
+    }
+  };
+
+  searchArticle = searchQuery => {
+    const { fetchResults: getResults, history, pageNumber } = {
+      ...this.props,
+      ...this.state
+    };
+    getResults(searchQuery, pageNumber, history);
+  };
+
+  handleTagFilter = (tag, index) => {
+    const { articles } = this.props;
+    const { activeTag } = this.state;
+    if (index === activeTag) {
+      return this.setState({
+        activeTag: null,
+        articles
+      });
+    }
+    this.setState({
+      articles: filterByTag(articles, tag),
+      activeTag: index
+    });
+  };
+
+  renderTags = (tagsList, activeTag) => (
+    <div>
+      <h2>#Tags</h2>
+      <div className="tags-section">
+        {tagsList.map((title, index) => (
+          <BasicButton
+            className={`tags ${activeTag === index ? "active" : ""}`}
+            title={title}
+            key={`${index + 1}`}
+            onClick={() => this.handleTagFilter(title, index)}
+            data-test="single-tag"
+          />
+        ))}
+      </div>
+    </div>
+  );
+
+  renderAuthors = authors => {
+    const authorsArr = values(authors);
+    return (
+      <div>
+        <h2>Authors</h2>
+        <div className="authors-section">
+          {authorsArr.map(author => (
+            <div key={author.id}>
+              <AuthorCard {...author} />
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  };
+
+  renderArticles = (articles, isLoading, errors, activeTag) => (
+    <section id="articles" className="active">
+      {values(articles).map(article => (
+        <div className="col-md-12" key={article.id}>
+          <Card
+            title={article.title}
+            description={article.description}
+            comments={article.comments.length}
+            likes={article.likes.length}
+            author={article.author}
+            readTime={article.readTime}
+            createdAt={calculateTimeStamp(article.createdAt)}
+            slug={article.slug}
+          />
+        </div>
+      ))}
+      <div className="loading-anim">
+        {isLoading ? (
+          <Loading />
+        ) : (
+          !isEmpty(errors.message) &&
+          !activeTag && (
+            <small data-test="error-message">{errors.message}</small>
+          )
+        )}
+      </div>
+    </section>
+  );
+
+  render() {
+    const {
+      searchQuery,
+      isLoading,
+      errors,
+      articles,
+      authors,
+      activeTag,
+      tagsList
+    } = { ...this.props, ...this.state };
+
+    return (
+      <div className="search-results">
+        <div className="left-side">
+          <div className="search-query">
+            <Input
+              type="search"
+              name="search"
+              placeholder=""
+              className="search-input"
+              onChange={e => this.handleOnChange(e.target.value)}
+              value={searchQuery}
+              onKeyDown={e => this.handleEnterPress(e)}
```

Comment:
Should be the function reference instead.

Reply:
<img width="944" alt="Screen Shot 2019-05-07 at 10 58 21" src="https://user-images.githubusercontent.com/20681465/57287107-13801d00-70b7-11e9-9f20-f45645066c6a.png">